lib/IDE/Refactoring.cpp

```diff
@@ -2066,6 +2067,160 @@ bool RefactoringActionSimplifyNumberLiteral::performChange() {
   return true;
 }
 
+static CallExpr *findTrailingClosureTarget(SourceManager &SM,
+                                           ResolvedCursorInfo CursorInfo) {
+  if (!CursorInfo.ContextNode)
+    // Middle of nowhere.
+    return nullptr;
+
+  SourceLoc Loc;
+  switch (CursorInfo.Kind) {
+  case CursorInfoKind::ExprStart:
+    Loc = CursorInfo.TrailingExpr->getStartLoc();
+    break;
+  case CursorInfoKind::StmtStart:
+    return nullptr;
+  default:
+    Loc = CursorInfo.Loc;
+    break;
+  }
```

Comment:
Can we just replace this with an early return in the second case? and use `SourceLoc Loc = CursorInfo.Loc;` in general. `CursorInfo.TrailingExpr->getStartLoc()` and `CursorInfo.Loc` should be equivalent.